src/Aspire.Hosting/MongoDB/MongoDbBuilderExtensions.cs

```diff
@@ -0,0 +1,92 @@
+    public static IResourceBuilder<MongoDbContainerResource> AddMongoDbContainer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int port = DefaultContainerPort,
+        string password = DefaultPassword)
+    {
+        var mongoDbContainer = new MongoDbContainerResource(name, password);
+
+        return builder
+            .AddResource(mongoDbContainer)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation(WriteMongoDbContainerToManifest))
+            .WithAnnotation(new ServiceBindingAnnotation(ProtocolType.Tcp, port: port, containerPort: DefaultContainerPort)) // Internal port is always 27017.
+            .WithAnnotation(new ContainerImageAnnotation { Image = "mongo", Tag = "latest" })
+            .WithEnvironment(PasswordEnvVarName, () => mongoDbContainer.Password);
+    }
+
+    /// <summary>
+    /// Adds a MongoDB connection to the application model. Connection strings can also be read from the connection string section of the configuration using the name of the resource.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="connectionString">The MongoDB connection string (optional).</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MongoDbConnectionResource}"/>.</returns>
+    public static IResourceBuilder<MongoDbConnectionResource> AddMongoDbConnection(this IDistributedApplicationBuilder builder, string name, string? connectionString = null)
+    {
+        var mongoDbConnection = new MongoDbConnectionResource(name, connectionString);
+
+        return builder
+            .AddResource(mongoDbConnection)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation((json) => WriteMongoDbConnectionToManifest(json, mongoDbConnection)));
+    }
+
+    /// <summary>
+    /// Adds a MongoDB database to the application model.
+    /// </summary>
+    /// <param name="builder">The MongoDB server resource builder.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MongoDbDatabaseResource}"/>.</returns>
+    public static IResourceBuilder<MongoDbDatabaseResource> AddDatabase(this IResourceBuilder<MongoDbContainerResource> builder, string name)
+    {
+        var mongoDbDatabase = new MongoDbDatabaseResource(name, builder.Resource);
+
+        return builder.ApplicationBuilder
+            .AddResource(mongoDbDatabase)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation(
+                (json) => WriteMongoDbDatabaseToManifest(json, mongoDbDatabase)));
+    }
+
+    private static void WriteMongoDbContainerToManifest(Utf8JsonWriter jsonWriter)
```

Comment:
We've just made a breaking API change which will impact you here. Specifically instead of piping through the JsonWriter you'll want to pipe through the `ManifestPublishingContext` which exposes a `Writer` property.

Reply:
I've changed to use the new method, please review.